### ISSUE TITLE
std.mem.asBytes error: expected single item pointer, passed usize

### ISSUE THREAD

**Issue by mikdusan** [ziglang/zig#14011]
### Zig Version

0.11.0-dev.900+0fb53bd24

#### z0.zig
```rust
test {
    const i: usize = 0;
    _ = @import("std").mem.asBytes(i);
}
```

#### shell
```
archlinux▸ zig test z0.zig
/home/mike/project/zig/work/main/lib/std/mem.zig:3373:9: error: expected single item pointer, passed usize
        @compileError("expected single item pointer, passed " ++ @typeName(P));
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mike/project/zig/work/main/lib/std/mem.zig:3381:47: note: called from here
pub fn asBytes(ptr: anytype) AsBytesReturnType(@TypeOf(ptr)) {
                             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
thread 11468 panic: Zig compiler bug: attempted to destroy declaration with an attached error
Analyzing z0.zig: z0.zig:test_0
      %3 = dbg_block_begin()
      %4 = dbg_stmt(2, 5)
      %5 = as_node(@Zir.Inst.Ref.usize_type, @Zir.Inst.Ref.zero)
      %6 = dbg_var_val(%5, "i")
      %7 = dbg_stmt(3, 5)
      %8 = import("std")
      %9 = ref(%8)
      %10 = dbg_stmt(3, 23)
      %11 = field_ptr(%9, "mem")
      %12 = dbg_stmt(3, 27)
      %13 = field_call_bind(%11, "asBytes")
      %14 = dbg_stmt(3, 35)
    > %15 = call(.auto, %13, [
        {
          %16 = break_inline(%15, %5)
        },
      ])
      %18 = dbg_block_end()
      %17 = ensure_result_non_error(%15)
      %19 = restore_err_ret_index(%2, @Zir.Inst.Ref.none)
      %20 = break(%2, @Zir.Inst.Ref.void_value)
    For full context, use the command
      zig ast-check -t z0.zig

  in z0.zig: z0.zig:test_0
    > %2 = block({%3..%20})

/home/mike/project/zig/work/main/src/Module.zig:3528:13: 0x33f4c87 in destroyDecl (zig)
            @panic("Zig compiler bug: attempted to destroy declaration with an attached error");
            ^
/home/mike/project/zig/work/main/src/Sema.zig:7120:33: 0x412f6e5 in instantiateGenericCall (zig)
        errdefer mod.destroyDecl(new_decl_index);
                                ^
/home/mike/project/zig/work/main/src/Sema.zig:6383:40: 0x3dd42e7 in analyzeCall (zig)
        if (sema.instantiateGenericCall(
                                       ^
/home/mike/project/zig/work/main/src/Sema.zig:6162:42: 0x399d9e6 in zirCall (zig)
            break :b try sema.analyzeCall(block, func, func_src, call_src, modifier, ensure_result_used, resolved_args, bound_arg_src);
                                         ^
/home/mike/project/zig/work/main/src/Sema.zig:916:62: 0x3767808 in analyzeBodyInner (zig)
            .call                         => try sema.zirCall(block, inst),
                                                             ^
/home/mike/project/zig/work/main/src/Sema.zig:5290:34: 0x3e14799 in resolveBlockBody (zig)
        if (sema.analyzeBodyInner(child_block, body)) |_| {
                                 ^
/home/mike/project/zig/work/main/src/Sema.zig:5273:85: 0x3a6325d in zirBlock (zig)
    return sema.resolveBlockBody(parent_block, src, &child_block, body, inst, &label.merges);
                                                                                    ^
/home/mike/project/zig/work/main/src/Sema.zig:1453:69: 0x3777037 in analyzeBodyInner (zig)
                if (!block.is_comptime) break :blk try sema.zirBlock(block, inst);
                                                                    ^
/home/mike/project/zig/work/main/src/Sema.zig:798:30: 0x39502e8 in analyzeBody (zig)
    _ = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                             ^
/home/mike/project/zig/work/main/src/Module.zig:5697:43: 0x3749efa in analyzeFnBody (zig)
    sema.analyzeBody(&inner_block, fn_info.body) catch |err| switch (err) {
                                          ^
/home/mike/project/zig/work/main/src/Module.zig:4377:40: 0x3577f0d in ensureFuncBodyAnalyzed (zig)
            var air = mod.analyzeFnBody(func, sema_arena) catch |err| switch (err) {
                                       ^
/home/mike/project/zig/work/main/src/Compilation.zig:3119:42: 0x3575a8c in processOneJob (zig)
            module.ensureFuncBodyAnalyzed(func) catch |err| switch (err) {
                                         ^
/home/mike/project/zig/work/main/src/Compilation.zig:3057:30: 0x343dc44 in performAllTheWork (zig)
            try processOneJob(comp, work_item);
                             ^
/home/mike/project/zig/work/main/src/Compilation.zig:2361:31: 0x3439748 in update (zig)
    try comp.performAllTheWork(main_progress_node);
                              ^
/home/mike/project/zig/work/main/src/main.zig:3437:20: 0x3468acd in updateModule (zig)
    try comp.update();
                   ^
/home/mike/project/zig/work/main/src/main.zig:3099:17: 0x3348b4d in buildOutputType (zig)
    updateModule(gpa, comp, hook) catch |err| switch (err) {
                ^
/home/mike/project/zig/work/main/src/main.zig:263:31: 0x3322f94 in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .zig_test);
                              ^
/home/mike/project/zig/work/main/src/main.zig:201:20: 0x3322405 in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/mike/project/zig/work/main/lib/std/start.zig:614:37: 0x3324b27 in main (zig)
            const result = root.main() catch |err| {
                                    ^
zsh: IOT instruction (core dumped)  zig test z0.zig

### Expected Behavior

normal `@compileError` to be emitted without "zig compiler bug" panic

**Comment by nektro** [comment]
you're likely looking for `std.mem.toBytes()`

**Comment by nektro** [comment]
crash is duplicate of https://github.com/ziglang/zig/issues/12927

**Comment by Vexu** [comment]
Duplicate of  #12927

This issue is well known which is why the panic is there. It's fixed by #13017 but it causes a memory leak that I've not yet managed to fix.
